uyuni-tools/last_update_export.py

```diff
@@ -0,0 +1,103 @@
+this creates a more streamlined export/import for Disconnected or Airgapped systems.
+
+Instructions:
+1. Ensure Python 3.x is installed on your system.
+2. This script must be run with root privileges to manage file permissions and perform system-level operations.
+3. Before running the script, update the `/root/.mgr-sync` configuration file with the correct manager login credentials using `mgr-sync -s refresh` to create the credentials file.
+4. Customize the directory path, by default `/mnt` was chosen but this could be any location you want, ensure the location has ample free space.
+5. Customize the 'rsync_user' and 'rsync_group' in the script to match the user and group names on your system.
+6. Schedule this script using a cron job or another scheduler for daily execution.
+
+Intended Usage:
+- Scheduled daily exports of software channel data.
+- Logging of export operations in '/mnt/logs'.
+- Exports are stored in '/mnt/export/updates'.
+- This script is intended for systems administrators managing software channel updates for Airgapped Environments.
+
+Ensure the server and paths are correctly configured and accessible before running the script.
+"""
+
+import os
+import subprocess
+import datetime
+from xmlrpc.client import ServerProxy
+import ssl
+import socket
+import configparser
+import shutil
+
+def setup_directories(base_path, output_path, log_path):
+    if not os.path.exists(base_path):
+        os.makedirs(base_path, exist_ok=True)
+    if not os.path.exists(log_path):
+        os.makedirs(log_path, exist_ok=True)
+    if os.path.exists(output_path):
+        shutil.rmtree(output_path)
+    os.makedirs(output_path, exist_ok=True)
+
+def setup_logging(log_dir, today):
+    log_file_path = os.path.join(log_dir, f"{today}-daily_export.log")
+    return log_file_path
+
+def create_client():
+    config_path = os.path.expanduser('/root/.mgr-sync')
+    config = configparser.ConfigParser()
+    with open(config_path, 'r') as f:
+        config.read_string('[DEFAULT]\n' + f.read())
+    MANAGER_LOGIN = config.get('DEFAULT', 'mgrsync.user')
+    MANAGER_PASSWORD = config.get('DEFAULT', 'mgrsync.password')
+    SUMA_FQDN = socket.getfqdn()
+    MANAGER_URL = f"https://{SUMA_FQDN}/rpc/api"
+    context = ssl.create_default_context()
+    client = ServerProxy(MANAGER_URL, context=context)
+    return client, client.auth.login(MANAGER_LOGIN, MANAGER_PASSWORD)
+
+# Configuration Variables
+base_dir = "/mnt"  # Define base directory where the exports will be.
+output_dir = os.path.join(base_dir, "export/updates")
+log_dir = os.path.join(base_dir, "logs")
+today = datetime.date.today()
+target_date = today - datetime.timedelta(days=1)  # Define the number of days back for export, 1 day by default
+rsync_user = "rsyncuser"  # Define rsync user
+rsync_group = "users"     # Define rsync group
```

Comment:
OTOH these ones I would move them to the top of the file (below imports statements and before functions definition), as user is expected to modify them.

I'd also suggest to use uppercase notation (constants) for those.